### ISSUE TITLE
[EPIC] Material 3 drawer

### ISSUE THREAD

**Issue by wmontwe** [thunderbird/thunderbird-android#8100]
The navigation drawer should look and feel consistent with the rest of the K-9 Mail and Thunderbird for Android Material 3 theme.
As the existing drawer is based on [MaterialDrawer](https://github.com/mikepenz/MaterialDrawer) it has limited Material 3 support and doesn't allow us to easily adapt to the new design. Therefore we will rewrite the navigation drawer from scratch using [Jetpack Compose](https://developer.android.com/compose).

Tasks:

- https://github.com/thunderbird/thunderbird-android/issues/8032
- https://github.com/thunderbird/thunderbird-android/issues/8103
- https://github.com/thunderbird/thunderbird-android/issues/8104
- https://github.com/thunderbird/thunderbird-android/issues/8118
- https://github.com/thunderbird/thunderbird-android/issues/8120
- https://github.com/thunderbird/thunderbird-android/issues/8121
- https://github.com/thunderbird/thunderbird-android/issues/8122
- https://github.com/thunderbird/thunderbird-android/issues/8123
- https://github.com/thunderbird/thunderbird-android/issues/8142
- https://github.com/thunderbird/thunderbird-android/issues/8164 once new drawer works as expected

To improve on:

- https://github.com/thunderbird/thunderbird-android/issues/8146
- Save scroll position
- Starred and unread count need to be updated instantly - https://github.com/thunderbird/thunderbird-android/issues/8765
- Folder names should limited to 2 lines and shortened otherwise - https://github.com/thunderbird/thunderbird-android/issues/8738
- Image loading needs to be implemented (no contact picture) -> needs design first
- Feature could be split into api and implementation module
- https://github.com/thunderbird/thunderbird-android/issues/8739
- https://github.com/thunderbird/thunderbird-android/issues/7556
- Ellipsize folder name in the middle: [support expected for compose foundation 1.8](https://issuetracker.google.com/issues/185418980)
- #8300

Related bugs:

- https://github.com/thunderbird/thunderbird-android/issues/8137
- Unread and star count are not properly updated for unified folders (old and new implementation)
- #8204
- #8301
- #8381
- https://github.com/thunderbird/thunderbird-android/issues/8487
- https://github.com/thunderbird/thunderbird-android/issues/8548
- https://github.com/thunderbird/thunderbird-android/issues/8709

**Comment by wmontwe** [comment]
The exsiting drawer has a complex setup to communicate account and folder changes between itself, the MessageList, MessageFragment and different entry points (widget, notifications). To simplify the communication flow we decided to introduce a separate view model to keep track of the currently active account and when needed selected folder.  It should be bound to the lifecycle of the MessageList activity and exposed to navigation drawer as source of truth.

**Comment by kewisch** [comment]
@marcRDZ You were asking if you could support on any other issues related to the drawer. @wmontwe is working on some of the foundations for future support of account pictures, once he is done with that there would be an opportunity to help on the image picker.

There are a few remaining issues left on this epic, are there any that are interesting to you? We might not have linked all the issues after the fact, so good to check functionality as well.

**Comment by marcRDZ** [comment]
@kewisch count on it, I'll review the list to check possible candidates to tackle asap

**Comment by marcRDZ** [comment]
Hi @kewisch, 

I'd been reviewing this a little bit and I have some questions...
 
>  [...]
> 
> To improve on:
> 
>   * Save scroll position (No idea what does it mean)
> 
>   * Starred and unread count need to be updated instantly (I guess is the same as second bug below)
> 
>   * Feature could be split into api and implementation module (May be too much for me to handle at this moment)
> 
> 
> Related bugs:
> 
>   * Unread and star count are not properly updated for unified folders (old and new implementation) (I saw a weird behaviour with badges only updating when switching between  accounts but I'm not sure if that's the matter)
> 
> 



So I think some good issues to resolve easily are:
   * Folder names should limited to 2 lines and shortened otherwise
   * Account name should be single line if email and display name are identical
   * Ellipsize folder name in the middle: [support expected for compose foundation 1.8](https://issuetracker.google.com/issues/185418980)
Should I create new issues for these?


Also I comment on #8548 a possible cause for the crash, Idk if it could have some sense for you but I'll try to review again on some moment

**Comment by wmontwe** [comment]
Hi @marcRDZ,

please go ahead and you could create the tickets, I'll link them here.

**Comment by marcRDZ** [comment]
Hi @wmontwe 

Here we've the issues:

#8738 
 * Folder names should limited to 2 lines and shortened otherwise
 * Ellipsize folder name in the middle: [support expected for compose foundation 1.8](https://issuetracker.google.com/issues/185418980)

#8739 
 * Account name should be single line if email and display name are identical

**Comment by shamim-emon** [comment]
@wmontwe @kewisch I have created [Show/Hide star count does not update immediately](https://github.com/thunderbird/thunderbird-android/issues/8765#issue-2803300794) issue. Seems like this should be part of this EPIC? If possible please have a look and confirm. Thanks

**Comment by wmontwe** [comment]
> [@wmontwe](https://github.com/wmontwe) [@kewisch](https://github.com/kewisch) I have created [Show/Hide star count does not update immediately](https://github.com/thunderbird/thunderbird-android/issues/8765#issue-2803300794) issue. Seems like this should be part of this EPIC? If possible please have a look and confirm. Thanks

@shamim-emon Thanks, added it

**Comment by shamim-emon** [comment]
@wmontwe @kewisch, created another issue [Show/Hide Unified Inbox does not update immediately](https://github.com/thunderbird/thunderbird-android/issues/8767) . I suppose this too should be part of this EPIC? If possible have a look and confirm.
Thank you!

**Comment by wmontwe** [comment]
I moved open issues to #8812: Account Drawer Improvements, please follow further work there.
